dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/JSONToolTest.java

```diff
@@ -0,0 +1,321 @@
+package com.dotcms.rendering.velocity.viewtools;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test {@link JSONTool}
+ * @author jsanca
+ */
+public class JSONToolTest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception{
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: {@link JSONTool#generate(String)}
+     * Given Scenario: Parsing an plain object
+     * ExpectedResult: Expected a map with the properties
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void test_Generate_simple_Map() throws Exception {
+
+        final JSONTool jsonTool   = new JSONTool();
+        final String   stringJson = "{\n" +
+                "            \"body\": \"## Container: Blank Container\\r\\n## This is autogenerated code that cannot be changed\\r\\n#parseContainer('d71d56b4-0a8b-4bb2-be15-ffa5a23366ea','1539784124854')\\r\\n\",\n" +
+                "            \"canPublish\": true,\n" +
+                "            \"canRead\": true,\n" +
+                "            \"canWrite\": true,\n" +
+                "            \"categoryId\": \"9d11b327-9bf5-4d42-92bd-7894d56ed218\",\n" +
+                "            \"countAddContainer\": 0,\n" +
+                "            \"countContainers\": 0,\n" +
+                "            \"deleted\": false,\n" +
+                "            \"drawed\": false,\n" +
+                "            \"drawedBody\": null,\n" +
+                "            \"footer\": \"null\",\n" +
```

Comment:
Codacy found an issue: [The String literal "            \"footer\": \"null\",\n" appears 4 times in this file; the first occurrence is on line 52](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315721)